### PR TITLE
DET-617: Build output and add tooling

### LINKED ISSUE #10: Too many console.log calls
The console.log calls in this library cause unnecessary noise in my logs, particularly if I'm calling a function lots of times.  Please can they be off by default?  (Perhaps add a config option to enable them if some people actually want them)

### PR DESCRIPTION
**Pull request by Limess**
* Added prettier/linter as per other de-tooling projects, fixed all errors as a result of this addition.

* Replaced built in polyfills with ponyfills/a thrown error in the case of es6-promise (this should only occur on the browser in which case the consumer should be responsible).

* Bundled in browser (umd), module (ecmascript modules) and node (cjs) formats using rollup, this runs through babel so the source can now use up to date syntax. This should now work just fine in browser environments. See https://philipwalton.com/articles/deploying-es2015-code-in-production-today/ for details. This project now uses node 8 but builds down to the previous target of 4.3.1.

* Added a test framework ([jest](https://facebook.github.io/jest/)) which as basic tests for functionality (not currently query strings, error scenarios) using jest snapshots to check responses (these are pretty big in the repository, along with cmdb response fixtures!). 

* Generate docs with jsdoc with the intent of hosting github pages instead of static docs in readme

* Add coveralls

Additionally the tests verify that the bundle loads correctly in all environments. This could be flushed out with actual API calls, but this is tricky to stub in chrome and I didn't with to add real calls here.

**N.B.** the apparent 11k lines of changes are composed of 8k of `package-lock.json`, 1.5k lines of generated jest snapshots and 500 lines of JSON fixtures so are not as daunting as they seem.

Closes https://github.com/Financial-Times/cmdb.js/issues/10 by hiding logging behind a `verbose` flag.

### CODE REVIEW

**Comment by GeoffThorpeFT** [comment]
Rebased to include Rob's putRelationship and to move to v4

File: README.md [hunk 1]
```diff
@@ -12,40 +12,41 @@ Pull requests welcomed.
 
 ### Inital Setup
 Pass in your apikey to the library to authenticate requests:
-```
-var CMDBclass = require("cmdb.js");
-var cmdb = new CMDBclass({
+```js
+const Cmdb = require("cmdb.js");
+const cmdb = new Cmdb({
```
Comment (lucas42) [inline README.md]: Is it confusing for people reading the examples to have 2 constants who name is the same except the case?

**Comment by lucas42** [comment]
I've not done a great deal with nodejs tooling recently, so probably not the best person to review this.  But it looks broadly fine to me.

**Comment by Limess** [comment]
~This has a couple of issues still, adding a test suite on this branch.~

Added tests.

File: README.md (hunk 1 above)
Comment (Limess) [reply README.md]: I don't think so. This is a standard pattern in most OO languages (upper case constructor, lower case instance).

Comment (lucas42) [reply README.md]: 👍

**Review by lucas42 — APPROVED** [review]

**Pull request merged**